### PR TITLE
h264e: change ICQ quality factor range to [1, 51].

### PR DESCRIPTION
**Pull request by ChenXiaomin36**
Change the ICQ quality factor range to [1, 51]
which aligns with the definition in VAAPI.

Change-Id: I582753964de57d0b479c1eebee42b05700f516d1

### CODE REVIEW

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 1]
```diff
@@ -2304,6 +2304,7 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
             par.mfx.RateControlMethod != MFX_RATECONTROL_AVBR &&
             par.mfx.RateControlMethod != MFX_RATECONTROL_QVBR &&
             par.mfx.RateControlMethod != MFX_RATECONTROL_VCM &&
+            par.mfx.RateControlMethod != MFX_RATECONTROL_ICQ &&
```
Comment (lizhong1008) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: So, ICQ mode can be supported by Lowpower mode  too?

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 2]
```diff
@@ -3854,8 +3855,11 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
         changed = true;
     }
 
-    if (par.mfx.RateControlMethod == MFX_RATECONTROL_ICQ ||
-        par.mfx.RateControlMethod == MFX_RATECONTROL_LA_ICQ)
+    if (par.mfx.RateControlMethod == MFX_RATECONTROL_ICQ)
+    {
+        if (!CheckRange(par.mfx.ICQQuality, 1, 51)) changed = true;
+    }
+    if (par.mfx.RateControlMethod == MFX_RATECONTROL_LA_ICQ)
```
Comment (lizhong1008) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: Why we need such a change?

**Comment by dvrogozh** [comment]
pr #866 will take care of the enabling icq in sample_encode...

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 1 above)
Comment (ChenXiaomin36) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: ICQ mode can't be supported by Lowpower mode now, so I remove this code. @lizhong1008

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 2 above)
Comment (ChenXiaomin36) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: In VAAPI, the comment for ICQ_quality_factor is that "Initial quality factor used in ICQ mode." and "This value must be between 1 and 51.". So I think in MSDK, the value check for it should align with the definition in VAAPI. @lizhong1008

**Comment by ChenXiaomin36** [comment]
The purpose of this change is to enable ICQ for AVC in sample_encode. I have removed the codes for ICQ support for Lowpower mode. @dvrogozh

**Comment by ChenXiaomin36** [comment]
Just see you submit the same PR. @dvrogozh

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 2 above)
Comment (ChenXiaomin36) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: If MSDK still checks the value in the range of [0, 51], and doesn't have process for 0, then 0 can be transferred to media driver. At present, the valid range for ICQ quality factor in media driver is [1, 51], if 0 is set, then media driver will report error. Currently, I haven't seen there is process for 0. If I missed this process, please correct me. Thanks! @dvrogozh

**Comment by ChenXiaomin36** [comment]
If MSDK still checks the value in the range of [0, 51], and doesn't have process for 0, then 0 can be transferred to media driver. At present, the valid range for ICQ quality factor in media driver is [1, 51], if 0 is set, then media driver will report error. Currently, I haven't seen there is process for 0. If I missed this process, please correct me. Thanks! @dvrogozh

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 2 above)
Comment (stasefremov) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: @ogolikov, @takulako, please look.

**Comment by ogolikov** [comment]
0 - is default value for ICQQuality
MediaSDK sets ICQQuality to 26 in this case in SetDefaults function.

**Comment by lizhong1008** [comment]
> 0 - is default value for ICQQuality
> MediaSDK sets ICQQuality to 26 in this case in SetDefaults function.
Could you show me the code where set it to be 26?

**Comment by lizhong1008** [comment]
> 0 - is default value for ICQQuality
> MediaSDK sets ICQQuality to 26 in this case in SetDefaults function.

So I prefer iHD driver can take 0 as 26 too.  And application can have same setting for ICQ and ICQ_LA

**Comment by ogolikov** [comment]
mfx_h264_enc_common_hw.cpp: 5395   
 if (par.mfx.RateControlMethod == MFX_RATECONTROL_ICQ || par.mfx.RateControlMethod == MFX_RATECONTROL_LA_ICQ)
    {
        if (par.mfx.ICQQuality == 0)
            par.mfx.ICQQuality = 26;
    }

**Comment by ChenXiaomin36** [comment]
Since there is separate processing for "ICQQuality==0", cancel this PR.

**Pull request closed without merging**